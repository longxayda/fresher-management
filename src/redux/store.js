import {configureStore} from "@reduxjs/toolkit";
import attendTableOneTraineeSlice from "./attendanceManagementSlice/attendTableOneTraineeSlice";

// import ticketClassSlice from "./attendanceManagementSlice/ticketClassSlice";
import diligentSummarySlice from "./attendanceManagementSlice/diligentSummarySlice";
import leaveRequestSlice from "./attendanceManagementSlice/leaveRequestSlice";
import checkboxFRReducer from "./fresherManageSlice/checkboxFresherSlice";
import evaluationSlice from "./auditorsSlice/evaluationSlice";
import resourceSlice from "./auditorsSlice/resourceSlice";
import auditorSlice from "./auditorsSlice/auditorSlice";
import manageFRReducer from "./fresherManageSlice/fresherManagementSlice";
import bonusPenaltyReducer from "./fresherManageSlice/penaltySlice";

import classSlice from "./classManagementSlice/classSlice";
import classDetailsSlice from "./classManagementSlice/classDetailsSlice";
import learningPathSlice from "./classManagementSlice/learningPathSlice";
import syllabusSlice from "./classManagementSlice/syllabusSlice";
import moduleSlice from "./classManagementSlice/moduleSlice";
import topicSlice from "./classManagementSlice/topicSlice";

import scoreSlice from "./fresherManageSlice/scoreSlice";
import topicMarkSlice from "./fresherManageSlice/topicMarkSlice";
import scheduleSlice from "./auditorsSlice/scheduleSlice";
import createNewAuditSlice from "./auditorsSlice/createNewAuditSlice";
import adminManageAttendanceReducer from "./attendanceManagementSlice/adminManageAttendanceSlice";
import adminTicketReducer from "./attendanceManagementSlice/adminTicketSlice";

import traineeAttendanceTableSlice from "./attendanceManagementSlice/traineeAttendanceTableSlice";
import authSlice from "./authSlice/authSlice";
import roleSlice from "./roleSlice";
import UserListSlice from "./UserListSlices/UserListSlice";
import attendanceSlice from "./attendanceManagementSlice/attendanceSlice";
import traineeUserProfileSlice from "./userProfileSlice/traineeUserProfileSlice";
import adminUserProfileSlice from "./userProfileSlice/adminUserProfileSlice";

import AddNewReportTableSlice from "components/WeeklyReportTables/addNewReportTableSlice";
import DetailReportTableSlice from "components/WeeklyReportTables/detailReportTableSlice";
import ReportListTableSlice from "components/WeeklyReportTables/reportListTableSlice";
import searchSlice from "components/Search/searchSlice";
import ReportModalSlice from "components/ReportModal/reportModalSlice";

import HistoryReportListTableSlice from "components/MonthlyReportTables/historyReportListTableSlice";
import coursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/coursesReportSlice";
import financeReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/financeReportSlice";
import detailCoursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/detailCourseReportSlice";
import detailFinanceReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/detailFinanceReportSlice";

import DetailDeliveryTableSlice from "components/MonthlyReportTables/detailDeliveryTableSlice";
import DeliveryReportListTableSlice from "components/MonthlyReportTables/deliveryReportListTableSlice";
import NewDeliveryTableSlice from "components/MonthlyReportTables/newDeliveryTableSlice";

import OutputsReportTableSlice from "components/OutputsReportTables/outputsReportTableSlice";
import fresherReportSlice from "../pages/OutputsReportPage/fresher/fresherReportSlice";
import svttReportSlice from "../pages/OutputsReportPage/svtt/svttReportSlice";

import detailOutputsReportSlice from "../pages/OutputsReportPage/detailOutputsReportSlice";
import detailFresherReportSlice from "../pages/OutputsReportPage/fresher/detailFresherReportSlice";
import detailSVTTReportSlice from "../pages/OutputsReportPage/svtt/detailSVTTReportSlice";

import thunk from "redux-thunk";
const store = configureStore({
	reducer: {
		Filters: searchSlice,
		evaluation: evaluationSlice,
		resource: resourceSlice,
		manageFR: manageFRReducer,
		checkboxFR: checkboxFRReducer,
		attendTableOneTrainee: attendTableOneTraineeSlice,
		// traineeTabTable: TraineeTabTableSlice,
		class: classSlice,
		learningPath: learningPathSlice,
		syllabus: syllabusSlice,
		classDetails: classDetailsSlice,
		module: moduleSlice,
		topicModule: topicSlice,
		auth: authSlice,
		role: roleSlice,
		// UserList
		UserList: UserListSlice,
		schedule: scheduleSlice,
		newAudit: createNewAuditSlice,
		auditor: auditorSlice,

		// attendTable: attendTableSlice,
		// ticketClass: ticketClassSlice,
		diligentSummary: diligentSummarySlice,
		leaveRequest: leaveRequestSlice,
		adminManageAttendance: adminManageAttendanceReducer,
		adminTicket: adminTicketReducer,
		attendance: attendanceSlice,
		traineeAttendanceTable: traineeAttendanceTableSlice,
		bonusPenalty: bonusPenaltyReducer,
		score: scoreSlice,
		topic: topicMarkSlice,
		traineeUserProfile: traineeUserProfileSlice,
		adminUserProfile: adminUserProfileSlice,

		ReportList: ReportListTableSlice,
		AddNewReport: AddNewReportTableSlice,
		DetailReport: DetailReportTableSlice,
		Filters: searchSlice,
		ReportModal: ReportModalSlice,

		HistoryReportList: HistoryReportListTableSlice,
		CoursesReport: coursesReportSlice,
		FinanceReport: financeReportSlice,
		DetailCoursesReport: detailCoursesReportSlice,
		DetailFinanceReport: detailFinanceReportSlice,
		DetailDeliveryReport: DetailDeliveryTableSlice,
		DeliveryReportList: DeliveryReportListTableSlice,
		NewDeliveryReport: NewDeliveryTableSlice,

		OutputsReport: OutputsReportTableSlice,
		FresherReport: fresherReportSlice,
		SVTTReport: svttReportSlice,
		DetailFresherReport: detailFresherReportSlice,
		DetailSVTTReport: detailSVTTReportSlice,
		DetailOutputsReport: detailOutputsReportSlice,
	},
	middleware: [thunk],
});

export default store;
