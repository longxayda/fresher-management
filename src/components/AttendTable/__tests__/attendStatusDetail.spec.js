import React from 'react';
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, shallow } from 'enzyme';
import store from 'redux/store';
import { Provider } from "react-redux";
import { AttendTable } from 'components/AttendTable/index';
Enzyme.configure({ adapter: new Adapter() });

// describe('Test Attendance Details Attend Table', () => {
//     let component;

//     test('render attendance details attend table\'s component with EMPTY props', () => {
//         const data = [];
//         component = shallow(<AttendTable data = { data } />);
//         expect(component.find('.attend-table-content').exists()).toEqual(true);

//     });

//     test('render attendance details attendtable\'s component with NULL props', () => {
//         const data = null;
//         component = shallow(<AttendTable data = { data } />);
//         expect(component.find('.attend-table-content').exists()).toEqual(false);
//     });

//     test('render attendance details attend table\'s component with VALID props', () => {
//         const data = [
//             {
//                 trainee_id: 76586282,
//                 firstName: "neck-eyciv",
//                 lastName: "bead-1v2sq",
//                 monthlyAttendanceResponses: [
//                     {
//                         "date": "2022-06-01",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-02",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-03",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-06",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-07",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-08",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-09",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-10",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-11",
//                         "status": "P"
//                     },

//                     {
//                         "date": "2022-06-14",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-15",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-16",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-17",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-18",
//                         "status": "P"
//                     },

//                     {
//                         "date": "2022-06-21",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-22",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-23",
//                         "status": "P"
//                     },

//                     {
//                         "date": "2022-06-25",
//                         "status": "P"
//                     },

//                     {
//                         "date": "2022-06-28",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-29",
//                         "status": "P"
//                     },
//                     {
//                         "date": "2022-06-30",
//                         "status": "P"
//                     },

//                 ]
//         }];
//         component = shallow(<AttendTable data = { data } />);
//         expect(component.find('.attend-table-content').exists()).toEqual(true);
//     });

//     test('render attendance details attend table\'s component with INVALID ALL props', () => {
//         const data = 'temp';
//         component = shallow(<AttendTable data = {data } />);
//         expect(component.find('.attend-table-content').exists()).toEqual(false);
//     });
// })

describe('Test Attendance Details Page', () => {
    let component;

    beforeEach(() => {
        component = mount(<Provider store = { store }> < AttendTable /> </Provider>);
    });

    test('has the correct attend table class', () => {
        expect(component.find('.attend-table-content').exists()).toEqual(true);
    });

    test('has the correct select boxes class', () => {
        expect(component.find('.attendance-select-boxes').exists()).toEqual(true);
    });

    test('has the correct attend table container class', () => {
        expect(component.find('.attend-table-container').exists()).toEqual(true);
    });

    test('render Attendance Details Page\'s component correctly', () => {
        component = mount(<Provider store = { store }> <AttendTable /> </Provider>);
        expect(component.html()).toMatchSnapshot();
    });
})