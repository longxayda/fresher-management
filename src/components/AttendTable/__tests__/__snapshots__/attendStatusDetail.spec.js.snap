// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Attendance Details Page render Attendance Details Page's component correctly 1`] = `" <div class=\\"attend-table-content container\\"><div class=\\"d-flex align-items-center justify-content-between pb-2\\"><div class=\\"me-3\\"><div class=\\"attendance-select-boxes\\"><select class=\\"me-1\\" style=\\"width: 15rem; background-color: rgb(255, 255, 255); border: 1px solid #e3e3e3; border-radius: 4px; color: rgb(86, 86, 86); padding: 8px 12px; height: 40px; box-shadow: none;\\"><option value=\\"Choose class\\">Choose class</option></select><select style=\\"width: 10rem; background-color: rgb(255, 255, 255); border: 1px solid #e3e3e3; border-radius: 4px; color: rgb(86, 86, 86); padding: 8px 12px; height: 40px; box-shadow: none;\\"><option value=\\"Choose class\\">Choose month</option><option value=\\"null-null\\">null-null</option></select></div></div><div class=\\"d-flex align-items-center justify-content-between\\"><div class=\\"me-3\\"><label class=\\"mb-0\\">Search: </label></div><div class=\\"me-3\\"><input type=\\"search\\" class=\\"form-control\\" value=\\"\\"></div></div></div><div style=\\"overflow: auto;\\" class=\\"pt-2 row\\"><div class=\\"attend-table-container container\\"><pre class=\\"chorma\\"><table role=\\"table\\" class=\\"table table-bordered table-striped table-hover\\"><thead><tr role=\\"row\\"><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\">Data is loading or you need to choose class and month!</div></th></tr><tr role=\\"row\\"><th colspan=\\"1\\" role=\\"columnheader\\"><div class=\\"heading\\" style=\\"font-size: 16px; font-weight: 600; justify-content: center; text-align: center; width: 100%;\\"></div></th></tr></thead><tbody role=\\"rowgroup\\"></tbody></table></pre></div></div><div class=\\"row\\"><div class=\\"pagination d-flex align-items-center justify-content-center col\\"><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-3 btn btn-primary\\">&lt;&lt;</button><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-3 btn btn-primary\\">&lt;</button><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-3 btn btn-primary\\">&gt;</button><button type=\\"button\\" disabled=\\"\\" class=\\"btn-fill me-5 btn btn-primary\\">&gt;&gt;</button><span class=\\"me-3 ml-5\\">Page<strong>1 of 0</strong></span><span class=\\"me-3\\"> Go to page: </span><input style=\\"width: 100px;\\" type=\\"number\\" class=\\"me-3 form-control\\" value=\\"1\\"><div class=\\"\\"><select class=\\"form-control\\"><option value=\\"10\\">Show 10</option><option value=\\"20\\">Show 20</option><option value=\\"30\\">Show 30</option><option value=\\"40\\">Show 40</option><option value=\\"50\\">Show 50</option></select></div><br></div></div></div> "`;
